### ISSUE TITLE
Train the classifier tree

### ISSUE THREAD

**Issue by jad-b** [jad-b/matchwell#3]
## Background
Well, it won't be much of a hierarchical classifier if we don't train the hierarchy of classifiers.

Quick def: __targets__: List of lists of string labels. Each label has the full prefix of its ancestry, providing namespacing.

The confusing bit is re-shaping the targets (List[labels]) to fit the classifier to be trained. I believe the selection of direct child labels to the classifier will suffice.

### Questions 
* Performance
  * Will their be a significant performance hit for all the targets re-shaping? This is an _O(parents*Y)_ operation, as only parent nodes receive classifiers, and we'll need to iterate over the targets _Y_ for each parent. We could expand the labeling into a binary matrix of size (Y x |labels|) in a single pass, and do lookups from there.
  * Will the training steps _require_ parallelization to be tolerable? It's an obvious optimization, but will we need it this early in development?
* Clarify whether this is a _multi-class_ or _multi-label_ problem, and determine the exact format the underlying `sklearn.Classifier`s will be expecting for their `Y` values. 

## Proposal
See existing code for `HierarchicalClassifier.fit()` in [models.py] for the general suggested approach to training. 

To answer the questions related to performance, we need the training time of each classifier, as well as the total time spent re-shaping the target data.

[models.py]: https://github.com/jad-b/matchwell/blob/31006e84498f629141b7310d7e70e4e5962cd9e7/matchwell/models.py
